src/function/scalar/string/bar.cpp

```diff
@@ -0,0 +1,94 @@
+#include "duckdb/common/exception.hpp"
+#include "duckdb/common/operator/cast_operators.hpp"
+#include "duckdb/common/types/string_type.hpp"
+#include "duckdb/common/types/value.hpp"
+#include "duckdb/common/types/vector.hpp"
+#include "duckdb/common/unicode_bar.hpp"
+#include "duckdb/common/vector_operations/generic_executor.hpp"
+#include "duckdb/function/scalar/string_functions.hpp"
+
+namespace duckdb {
+
+static string_t BarScalarFunction(double x, double min, double max, double max_width, vector<char> &result) {
+	static const char *FULL_BLOCK = UnicodeBar::FullBlock();
+	static const char *const *PARTIAL_BLOCKS = UnicodeBar::PartialBlocks();
+	static const idx_t PARTIAL_BLOCKS_COUNT = UnicodeBar::PartialBlocksCount();
+	static const idx_t UNICODE_BLOCK_CHAR_SIZE = strlen(FULL_BLOCK);
```

Comment:
This won't work for the space character, no? As it is only a single byte.

Reply:
Removed `UNICODE_BLOCK_CHAR_SIZE`